File: tests/XMP/global-view/shadow-reflect/F/40.f90
program test

!$xmp nodes p(*)

!$xmp template t(64)
!$xmp distribute t(block) onto p

  integer :: a(64)
!$xmp align a(i) with t(i)
!$xmp shadow a(1)

  integer :: b(0:65)

  integer :: result = 0

!$xmp loop on t(i)
  do i = 1, 64
     a(i) = i
  end do

  b(0) = 64
  do i = 1, 64
     b(i) = i
  end do
  b(65) = 1

!$xmp reflect (a) width(/periodic/1:1)

!$xmp loop on t(i) reduction(+:result)
  do i = 1, 64

     if (a(i-1) /= b(i-1)) then
        result = 1
     end if

     if (a(i) /= b(i)) then
        result = 1
     end if

       if (a(i+1) /= b(i+1)) then
        result = 1
     end if

  end do

!$xmp task on p(1)
  if (result == 0) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
!$xmp end task

end program test

File: tests/XMP/global-view/shadow-reflect/C/40.c
#include <stdio.h>
#include <stdlib.h>

int main(){

#pragma xmp nodes p(*)

#pragma xmp template t(0:63)
#pragma xmp distribute t(block) onto p

  int a[64];
#pragma xmp align a[i] with t(i)
#pragma xmp shadow a[1]

  int b[66];

  int result = 0;

#pragma xmp loop on t(i)
  for (int i = 0; i < 64; i++){
    a[i] = i;
  }

  b[0] = 63;
  for (int i = 1; i < 65; i++){
    b[i] = i - 1;
  }
  b[65] = 0;

#pragma xmp reflect (a) width(/periodic/1:1)

#pragma xmp loop on t(i) reduction(+:result)
  for (int i = 0; i < 64; i++){

    if (a[i-1] != b[i]){
      result = 1;
    }

    if (a[i] != b[i+1]){
      result = 1;
    }

    if (a[i+1] != b[i+2]){
      result = 1;
    }

  }

#pragma xmp task on p(1)
  {
    if (result == 0){
      printf("PASS\n");
    }
    else {
      printf("ERROR\n");
      exit(1);
    }
  }

  return 0;
}
